harness/determined/common/experimental/determined.py

```diff
@@ -155,25 +155,23 @@ def create_model(
 
         return model.Model._from_json(r.json().get("model"), self._session)
 
-    def get_model(self, name: str) -> model.Model:
+    def get_model(self, identifier: Union[str, int]) -> model.Model:
         """
         Get the :class:`~determined.experimental.Model` from the model registry
-        with the provided name. If no model with that id is found in the registry,
+        with the provided name or ID. If no corresponding model is found in the registry,
         an exception is raised.
         """
-        r = self._session.get("/api/v1/models/{}".format(name))
-        return model.Model._from_json(r.json().get("model"), self._session)
+        r = self._session.get("/api/v1/models/{}".format(identifier)).json()
+        assert r.get("model", False)
+        return model.Model._from_json(r.get("model"), self._session)
 
     def get_model_by_id(self, model_id: int) -> model.Model:
```

Comment:
can you modify this docstring to show it is deprecated, and also include a deprecation warning like [this one](https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/checkpoint/_checkpoint.py#L224)?